ipykernel/kernelbase.py

```diff
@@ -859,6 +863,41 @@ async def debug_request(self, stream, ident, parent):
                                       parent, ident)
         self.log.debug("%s", reply_msg)
 
+    # Taken from https://github.com/jupyter-server/jupyter-resource-usage/blob/e6ec53fa69fdb6de8e878974bcff006310658408/jupyter_resource_usage/metrics.py#L16
+    def get_process_metric_value(self, process, name, attribute=None):
+        try:
+            # psutil.Process methods will either return...
+            metric_value = getattr(process, name)()
+            if attribute is not None:  # ... a named tuple
+                return getattr(metric_value, attribute)
+            else:  # ... or a number
+                return metric_value
+        # Avoid littering logs with stack traces
+        # complaining about dead processes
+        except BaseException:
+            return 0
+
+    async def usage_request(self, stream, ident, parent):
+        reply_content = {}
+        if psutil is None:
+            return reply_content
+        current_process = psutil.Process()
+        all_processes = [current_process] + current_process.children(recursive=True)
+        process_metric_value = lambda process, name, attribute: self.get_process_metric_value(
+                process, name, attribute
+        )
```

Comment:
is the lambda necessary, can't you just pass `self.get_process_metric_value` below ?

Reply:
That lambda function is used after by sum([process_metric_value(process, 'cpu_percent', None) for process in all_processes]) to sum the metric values on all the processes.

As it is written now, it is a sugar and is reused twice. I could rewrite that code block, it is just a matter of code styling I think.

Reply:
Ok, sorry if what i'm saying is unclear. 

What's the difference with:

```diff
- process_metric_value = lambda process, name, attribute: self.get_process_metric_value(
                process, name, attribute
        )
+ process_metric_value = self.get_process_metric_value
```

You don't capture any values, you don't cache, and don't change  arguments order.

Reply:
oh, true. It was clear but I have not seen the obvious simplification. Thx for helping here :) That change has been committed and works fine.

Reply:
I think there is a lambda calcul term for that is is a β-reduction ?

https://en.wikipedia.org/wiki/Lambda_calculus#%CE%B2-reduction

Reply:
> I think there is a lambda calcul term for that is is a β-reduction ?

I trust you on that :)